frontend/src/components/DataDashboard.vue

```diff
@@ -0,0 +1,130 @@
+<script setup lang="ts">
+import { ref, onMounted } from 'vue';
+
+const tableItems = ref(Array.from({ length: 5 }));
+const tableCols = ref();
+const gender = ref();
+
+tableCols.value = [
+  {
+    field: "date",
+    header: "Date"
+  },
+  {
+    field: "value",
+    header: "Population"
+  }
+]
+
+let apiCache = new ApiCache();
+
+async function getData(request : string) {
+  if (this.apiCache.recordExists(request)) {
+    tableItems.value = this.apiCache.get(request).observations;
+    console.log("Cache accessed");
+  }
+
+  else {
+    const res = await fetch(request);
+    const finalRes = await res.json();
+    tableItems.value = finalRes.observations;
+    this.apiCache.set(request, finalRes);
+    console.log("Cache set");
+  }
+}
+
+function genderToDcid(gender: string): string {
+  let dcidMap = new Map<string, string>([
+    ["Males", "Count_Person_25OrMoreYears_EducationalAttainmentBachelorsDegree_Male"],
+    ["Females", "Count_Person_25OrMoreYears_EducationalAttainmentBachelorsDegree_Female"]
+  ]);
+
+  let dcid = dcidMap.get(gender);
+  return dcid == undefined ? "" : dcid;
+}
+
+function buildRequest(dcid: string): string {
+  // This uses DataCommons' public API key. DO NOT INCLUDE A PRIVATE API KEY HERE!
+  return "https://api.datacommons.org/v1/observations/series/country/USA/" + dcid + "?key=AIzaSyCTI4Xz-UW_G2Q2RfknhcfdAnTHq5X5XuI";
+}
+
+</script>
+
+
+<template>
+  <span>Show US Population with a Bachelor's Degree for: {{ gender }}</span>
+
+  <div class="p-field-radiobutton">
+    <RadioButton id="gender" name="gender" value="Males" v-model="gender" />
+    <label for="gender">Males</label>
+  </div>
+  <div class="p-field-radiobutton">
+    <RadioButton id="gender2" name="gender" value="Females" v-model="gender" />
+    <label for="gender2">Females</label>
+  </div>
+
+  <!-- Debugging info 
+  <span> dcid to query: {{ genderToDcid(gender) }}</span>
+  <span> request: {{  buildRequest(genderToDcid(gender)) }} </span>
+  -->
+  {{ getData(buildRequest(genderToDcid(gender))) }}
```

Comment:
Actually, within your definition of `getData` we can remove the references to `this.` when we refer to the `apiCache`. 

sorry- that last comment was related to something else.

Reply:
Removing the `this.` references does remove the `Uncaught TypeError` . After that change, using `onMounted` at the end of the setup script instead of in the previous place (`{{ getData(buildRequest(genderToDcid(gender))) }}`) does not render any data:


![Screenshot 2023-07-18 12 57 54 PM](https://github.com/Spelman-College/spelman-dashboard/assets/13266735/53fa4654-839f-48b7-b334-81acb091420b)


WDYT?

Reply:
I think this is related to my previous comment about using `gender.value` in `onMounted`.

Reply:
Ack, this seems straightforward and I am following https://fjolt.com/article/vue-template-refs but the ref still doesn't seem properly attached to gender. Have to put this down for the day, will take a look when I next have bandwidth.

Reply:
See my [comment](https://github.com/Spelman-College/spelman-dashboard/pull/50/files#r1268387011) about swapping `onMounted` for `watchEffect`.